File: Practice/Stack/EvaluateExpression.js
function EvaluateExpression(A) {
  if (A.length == 1) {
    return A[0];
  }

  if (A.length == 2) {
    if (A[1] == `-`) {
      return -1 * A[0];
    }
    return A[0];
  }

  let stack = [];
  let ans = null;
  for (let i = 0; i < A.length; i++) {
    if (`+-*/`.includes(A[i])) {
      let val2 = stack.pop();
      let val1 = stack.pop();
      let operator = A[i];
      switch (operator) {
        case `+`:
          ans = Number(val1) + Number(val2);
          break;
        case `-`:
          ans = Number(val1) - Number(val2);
          break;
        case `*`:
          ans = Number(val1) * Number(val2);
          break;
        case `/`:
          ans = Math.floor(Number(val1) / Number(val2));
          break;
      }
      stack.push(ans);
    } else {
      stack.push(A[i]);
    }
  }

  return ans;
}

A = ["4", "13", "5", "/", "+"];
console.log(EvaluateExpression(A));

File: Revision/Recursion/EvaluateExpression.java
import java.util.Stack;

public class EvaluateExpression {
    public static void main(String[] args) {
        String[] A = {"2", "2", "/" };
        EvaluateExpression evaluateExpression = new EvaluateExpression();
        System.out.println(evaluateExpression.evalRPN(A));
    }
    public int evalRPN(String[] A) {
        Stack<String> s = new Stack<String>();
        for(int i=0;i<A.length;i++){
            String currentValue = A[i];

            if(currentValue.equals("+") || currentValue.equals("-") || currentValue.equals("*") || currentValue.equals("/")){
                int val1 = Integer.parseInt(s.pop());
                int val2 = Integer.parseInt(s.pop());
                int output = 0;
                switch(currentValue){
                    case "+":
                        output = val2 + val1;
                        break;
                    case "-":
                        output = val2 - val1;
                        break;
                    case "*":
                        output = val2 * val1;
                        break;
                    case "/":
                        output = val2/val1;
                        break;

                }

                s.push(Integer.toString(output));
            }
            else{
                s.push(A[i]);
            }
        }

        return Integer.parseInt(s.pop());
    }
}
